cpp/daal/src/algorithms/tsne/tsne_gradient_descent_fpt.cpp

```diff
@@ -365,21 +370,28 @@ services::Status summarizationKernelImpl(IdxType * count, IdxType * child, DataT
                         else
                         {
                             cnt += duplicates[ch];
-                            m = mass[ch] + DataType(duplicates[ch]) - 1.;
+                            m = mass[ch] + DataType(duplicates[ch]) - DataType(1);
                         }
                     }
                     else
                         m = (ch >= N) ? (cnt += count[ch], curMass[i]) : (cnt++, mass[ch]);
                     // add child's contribution
                     cm += m;
-                    px += posx[ch] * m;
-                    py += posy[ch] * m;
+                    //DAAL_OVERFLOW_CHECK_BY_MULTIPLICATION(DataType, posX[ch], m);
```

Comment:
Are you sure that posX[ch] * m won't overflow

Reply:
this is the setting for openmp parallelization. It was introduced in scikit-learn 1.1 and is always equal to 1 so far.

Reply:
One of the checks  broken python tests, I'm still investigate why